s3/s3_operations.cpp

```diff
@@ -416,6 +416,7 @@ namespace irods_s3 {
 
         if (oflag & O_APPEND) {
             mode |= ios_base::app;
+            mode &= ~ios_base::trunc;  // turn off trunc flag
```

Comment:
Just to be clear, this particular translation is an issue only for S3, right? Meaning, if `trunc` is set in the openmode, the open of the S3 object - not the iRODS replica - is what fails. Just making sure I understand

Reply:
This is an issue if we are using a cache file.  I don't remember exactly but I believe O_APPEND and O_TRUNC are allowed in POSIX but the combination of std::ios_base::append and std::ios_base::trunc are not allowed in streams.